dsp_permissions_scripts/doap/doap_model.py

```diff
@@ -12,47 +12,42 @@
 class Doap(BaseModel):
     """Model representing a DOAP, containing the target, the scope and the IRI of the DOAP."""
 
-    target: DoapTarget
+    target: GroupDoapTarget | EntityDoapTarget
     scope: PermissionScope
     doap_iri: str
 
 
-class DoapTarget(BaseModel):
+class GroupDoapTarget(BaseModel):
+    project_iri: str
+    group: Group
+
+
+class EntityDoapTarget(BaseModel):
     project_iri: str
-    group: Group | None = None
     resource_class: str | None = None
     property: str | None = None
 
     @model_validator(mode="after")
-    def assert_correct_combination(self) -> Self:
-        # asserts that DOAP is only defined for Group or ResourceClass or Property
-        # or a combination of ResourceClass and Property
-        match (self.group, self.resource_class, self.property):
-            case (None, None, None):
-                raise ValueError("At least one of group, resource_class or property must be set")
-            case (_, None, None) | (None, _, _):
-                pass
-            case _:
-                raise ValueError("Invalid combination of group, resource_class and property")
+    def _validate(self) -> Self:
+        if self.resource_class is None and self.property is None:
+            raise ValueError("At least one of resource_class or property must be set")
         return self
 
 
-class NewDoapTarget(BaseModel):
+class NewGroupDoapTarget(BaseModel):
```

Comment:
what is the difference between `GroupDoapTarget` and `NewGroupDoapTarget`? They seem pretty identical to me at first glance. Is this duplication really helpful?

Reply:
makes sense, maybe not nice, but pragmatic, so no reason to worry about it